test-prow-e2e.sh

```diff
@@ -0,0 +1,149 @@
+
+function createSecret {
+    oc create secret generic ${SECRET_NAME} --from-file=.dockerconfigjson=${PULL_SECRET_PATH} --type=kubernetes.io/dockerconfigjson -n $1
+}
+
+function linkSecrets {
+  for serviceAccount in `oc get serviceaccounts -n ${NS} --no-headers -o custom-columns=":metadata.name" | sed 's/"//g'`; do
+        echo "Linking ${SECRET_NAME} to ${serviceAccount}"
+        oc secrets link ${serviceAccount} ${SECRET_NAME} -n ${NS} --for=pull
+  done
+}
+
+function deleteAllPods {
+  oc delete pods --all -n $1 
+}
+
+echo "Creating secret for CI builds in ${NAMESPACE}"
+
+createSecret ${NAMESPACE}
+
+echo "Creating secret for linking pods"
+createSecret ${NS}
+
+echo "Adding secret to all service accounts in ${NS} namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods ${NS}
+
+sleep 30
+
+echo "Adding secret to all service accounts in ${NS} namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods ${NS}
+
+sleep 120
+
+echo "Adding secret to all service accounts in ${NS} namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods ${NS}
+
+sleep 120
+
+# Enable console plugin for nmstate
+export CONSOLE_CONFIG_NAME="cluster"
+export NMSTATE_PLUGIN_NAME="nmstate-console-plugin"
+NMSTATE_PLUGIN_IMAGE="$1"
+
+
+echo ""
+echo "Installing kubernetes-nmstate operator"
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/nmstate.io_nmstates.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/namespace.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/service_account.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/role.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/role_binding.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/operator.yaml
+
+echo "Deploy nmstate console plugin"
+oc process -f oc-manifest.yaml \
+  -p PLUGIN_NAME=nmstate-console-plugin \
+  -p NAMESPACE=openshift-nmstate \
+  -p IMAGE=$NMSTATE_PLUGIN_IMAGE \
+  | oc create -f -
+
+until \
+  oc wait pods -n ${NS} --for=jsonpath='{.spec.containers[0].image}'="$NMSTATE_PLUGIN_IMAGE" -l app=nmstate-console-plugin
+  do
+    sleep 1
+  done
+
+INSTALLER_DIR=${INSTALLER_DIR:=${ARTIFACT_DIR}/installer}
+
+BRIDGE_KUBEADMIN_PASSWORD="$(cat "${KUBEADMIN_PASSWORD_FILE:-${INSTALLER_DIR}/auth/kubeadmin-password}")"
+export BRIDGE_KUBEADMIN_PASSWORD
+BRIDGE_BASE_ADDRESS="$(oc get consoles.config.openshift.io cluster -o jsonpath='{.status.consoleURL}')"
+export BRIDGE_BASE_ADDRESS
+
+# Disable color codes in Cypress since they do not render well CI test logs.
+# https://docs.cypress.io/guides/guides/continuous-integration.html#Colors
+export NO_COLOR=1
+
+# Install dependencies.
+yarn install --ignore-engines
+
+# Run tests.
+yarn run cypress
```

Comment:
if you are using the operator, e.g. install on cluster, this should run on the cluster console